plugins/chat/assets/javascripts/discourse/components/chat/header/icon/unread-indicator.gjs

```diff
@@ -34,6 +34,12 @@ export default class ChatHeaderIconUnreadIndicator extends Component {
     );
   }
 
+  get hasUnreads() {
+    return (
+      this.unreadCount > 0 || this.chatTrackingStateManager.hasUnreadThreads
```

Comment:
This is the `gist` of the fix. There are unreads should also take into account threads.